File: src/main/java/nnglebanov/daoexample/domain/Genre.java
package nnglebanov.daoexample.domain;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Data
@EqualsAndHashCode(exclude = "books")
@ToString(exclude = "books")
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "GENRES")
public class Genre {
    @ManyToMany(mappedBy = "genres")
    List<Book> books;
    private String genreName;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id = 0;
    @CreationTimestamp
    private Date createdAt;
}

File: src/main/java/nnglebanov/daoexample/domain/Genre.kt
package nnglebanov.daoexample.domain


import nnglebanov.daoexample.helpers.Utils
import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.mongodb.core.mapping.MongoId
import java.util.*


@Document
data class Genre(
        @Field("genreName") val genreName: String,
        @DBRef val books: MutableList<Book>?,
        @Field("dateTime") val dateTime: Date) {
    @MongoId
    var id: String = Utils.generateId()
}
